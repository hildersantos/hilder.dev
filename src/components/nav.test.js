import React from "react"
import { render } from "react-testing-library"
import { StaticQuery, useStaticQuery } from "gatsby"

import Nav from "./nav"

import fixtureSiteInfo from "../../tests/__fixtures__/fixture-site-info"
import fixtureAllMarkdownRemark from "../../tests/__fixtures__/fixture-all-markdown-remark"

const props = {
  ...fixtureSiteInfo,
  ...fixtureAllMarkdownRemark,
}
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => render(props)),
    useStaticQuery.mockReturnValue(props)
})

describe("Nav", () => {
  it("renders without errors", () => {
    const { asFragment } = render(<Nav />)
    expect(asFragment()).toMatchSnapshot()
  })
})
