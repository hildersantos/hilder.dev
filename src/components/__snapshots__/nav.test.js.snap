// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders without errors 1`] = `
<DocumentFragment>
  <div
    class="navigation"
  >
    <ul
      class="navigation__list"
    >
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="test_0"
        >
          Test 0
        </a>
      </li>
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="test_1"
        >
          Test 1
        </a>
      </li>
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="test_2"
        >
          Test 2
        </a>
      </li>
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="test_3"
        >
          Test 3
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
